edalize/nextpnr.py

```diff
@@ -0,0 +1,124 @@
+import logging
+import os.path
+
+from edalize.edatool import Edatool
+from edalize.yosys import Yosys
+from importlib import import_module
+
+logger = logging.getLogger(__name__)
+
+class Nextpnr(Edatool):
+
+    argtypes = []
+
+    @classmethod
+    def get_doc(cls, api_ver):
+        if api_ver == 0:
+            yosys_help = Yosys.get_doc(api_ver)
+            nextpnr_help = {
+                "members": [
+                    {
+                        "name": "arch",
+                        "type": "String",
+                        "desc": "Target architecture. Legal values are *xilinx*",
```

Comment:
Is this backend only for Xilinx devices? What about iCE40 and ECP5 that also uses nextpnr?

Reply:
According to the readme from the official repository, `nextpnr-xilinx` is not in mainline. I think we can rename this script to `nextpnr_xilinx.py` since we assume to use the `nextpnr-xilinx` software which targets only xilinx platforms.

Reply:
Ok, let's rename it to nextpnr_xilinx. Then we can also remove all the arch stuff and just hard code xilinx (or AMD perhaps :) ) where needed

Reply:
I got rid of `arch` stuff